lib/her/model/attributes.rb

```diff
@@ -32,6 +32,7 @@ def initialize(attributes={})
         attributes = self.class.default_scope.apply_to(attributes)
         assign_attributes(attributes)
         yield self if block_given?
+        @changed_attributes = {} if persisted?
```

Comment:
Well spotted. About to head out the door. I think this may miss some cases but I have an idea that I will try on Monday. Thanks!

Reply:
I've tried to make this work but regardless of which approach I use, the `id` parameter is ignored here. I haven't figured out why yet and I'm very busy this week but hopefully I'll get time on Friday.

Reply:
Found the reason. In `use_setter_methods`, both `id` and the actual primary key name are treated as reserved so assignment methods are not created for them when the record is instantiated. They end up in the `unset_attributes` hash in the `assign_attributes` and simply get merged into the `@attributes` hash. This means that no change tracking is applied. If you directly assign new values using setter methods afterwards then change tracking does work because `method_missing` is invoked, which creates attribute methods without checking a reserved list first.

If I change `use_setter_methods` to not treat these as reserved then the above works but it breaks the `handles new resource with custom primary key` test. I need to find a way for the `id` method to not be redefined without blocking the creation of an `id=` method.